osu.Game/Online/API/Requests/Responses/APIScoreInfo.cs

```diff
@@ -15,9 +15,68 @@
 
 namespace osu.Game.Online.API.Requests.Responses
 {
-    public class APILegacyScoreInfo
+    public class APIScoreInfo : IScoreInfo
     {
-        public ScoreInfo CreateScoreInfo(RulesetStore rulesets)
+        [JsonProperty(@"score")]
+        public long TotalScore { get; set; }
+
+        [JsonProperty(@"max_combo")]
+        public int MaxCombo { get; set; }
+
+        [JsonProperty(@"user")]
+        public User User { get; set; }
+
+        public bool HasReplay { get; set; }
+
+        [JsonProperty(@"id")]
+        public long OnlineID { get; set; }
+
+        [JsonProperty(@"replay")]
+        public bool Replay { get; set; }
+
+        [JsonProperty(@"created_at")]
+        public DateTimeOffset Date { get; set; }
+
+        [JsonProperty(@"beatmap")]
+        public APIBeatmap Beatmap { get; set; }
+
+        [JsonProperty("accuracy")]
+        public double Accuracy { get; set; }
+
+        [JsonProperty(@"pp")]
+        public double? PP { get; set; }
+
+        [JsonProperty(@"beatmapset")]
+        public APIBeatmapSet BeatmapSet
+        {
+            set
+            {
+                // in the deserialisation case we need to ferry this data across.
```

Comment:
Looks like this is sort of relying on `Beatmap` to be deserialised first. I'd probably add a symmetric thing in `Beatmap` just for safety because I don't know what the guarantees wrt deserialisation order are if there are any.

Reply:
I don't believe I've changed this from how it's been until now. Not sure on the value of adding symmetry, but I guess we can if it's seen as beneficial.

Reply:
Fair enough then. Difficult to tell given how many pieces have moved.

Reply:
I have changed this into a throw, since we should never get to the scenario where the storage is the wrong type (or nullable).